libs/traits/src/lib.rs

```diff
@@ -549,8 +556,8 @@ pub trait TokenSwaps<Account> {
 		account: Account,
 		currency_in: Self::CurrencyId,
 		currency_out: Self::CurrencyId,
-		buy_amount: Self::Balance,
-		sell_rate_limit: Self::SellRatio,
+		amount_out: Self::Balance,
+		price: OrderPrice<Self::SellRatio>,
```

Comment:
These are the API changes:
- `amount_out` is the `sell_amount`. I've decided to call it `amount_out` to show that it's an amount measured in `currency_out`, but I'm open to suggestions.
- The `OrderPrice` enum defines how the price of the order will be computed.

Reply:
I am supporting `amount_out` to reduce confusion.

Reply:
But I think `price` is really `buy_price_limit`, right?

Reply:
Docs are not adapted.

Reply:
Yes, I need a final self-review updating all docs, thanks for the reminder

Reply:
I've modified the name because, as far as I can see, there is no "limit". You always buy/sell at the exact price you put there. A fixed price if `OrderPrice::Custom` is chosen, or the exact market price if `OrderPrice::market` is chosen.

If we want to add a limit order in the future, we should probably create another variant as `OrderPrice::Limit(price)`, which already holds the "limit" concept. WDYT?

Reply:
Which I should indicate probably is the price direction, if it is a `buy_price` or a `sell_price` 👍🏻

Reply:
As another thought, I was thinking of calling it `conversion_rate` as a rate to pass from `currency_out` to `currency_in.` I feel like buy/sell concepts here only make things hard to understand. WDYT?

Reply:
You are right. I think keeping `price` is a good choice. No blocker from my side.